apps/ledger-live-desktop/src/renderer/families/osmosis/AccountHeaderManageActions.js

```diff
@@ -26,20 +26,29 @@ const AccountHeaderActions = ({ account, parentAccount }: Props) => {
   const hasDelegations = cosmosResources.delegations.length > 0;
 
   const onClick = useCallback(() => {
-    if (hasDelegations) {
-      dispatch(
-        openModal("MODAL_OSMOSIS_DELEGATE", {
-          account,
-        }),
-      );
+    if (earnRewardEnabled) {
+      if (hasDelegations) {
+        dispatch(
+          openModal("MODAL_OSMOSIS_DELEGATE", {
+            account,
+          }),
+        );
+      } else {
+        dispatch(
+          openModal("MODAL_OSMOSIS_REWARDS_INFO", {
+            account,
+          }),
+        );
+      }
     } else {
       dispatch(
-        openModal("MODAL_OSMOSIS_REWARDS_INFO", {
+        openModal("MODAL_NO_FUNDS_STAKE", {
```

Comment:
Can you remind me, why are we opening the no-fund modal to a user that cannot delegate? 

I don't fully understand why we don't keep the Stake button disabled when they cannot delegate, and then we don't need to change the `onClick` to check for this earnRewardEnabled.